codespan-reporting/tests/term.rs

```diff
@@ -844,3 +844,77 @@ mod unicode_spans {
     test_emit!(rich_no_color);
     test_emit!(short_no_color);
 }
+
+mod multiline_omit {
+    use super::*;
+
+    #[test]
+    fn rich_no_color() {
+        let config = Config {
+            display_style: DisplayStyle::Rich,
+            context_lines: 1,
+            ..TEST_CONFIG.clone()
+        };
+
+        let mut files = SimpleFiles::new();
+
+        let file_id1 = files.add(
+            "empty_if_comments.lua",
+            [
+                "elseif 3 then", // primary label starts here
+                "",              // context line
+                "",
+                "",
+                "",
+                "",
+                "",
+                "",
+                "",     // context line
+                "else", // primary label ends here
+            ]
+            .join("\n"),
+        );
+
+        let file_id2 = files.add(
+            "src/lib.rs",
+            [
+                "fn main() {",
+                "    1",   // primary label starts here
+                "    + 1", // context line
+                "    + 1", // skip
+                "    + 1", // skip
+                "    + 1", // skip
+                "    +1",  // secondary label here
+                "    + 1", // this single line will not be skipped; the previously filtered out label must be retrieved
+                "    + 1", // context line
+                "    + 1", // primary label ends here
+                "}",
+            ]
+            .join("\n"),
+        );
+
+        let diagnostics = vec![
+            Diagnostic::error()
+                .with_message("empty elseif block")
+                .with_code("empty_if")
+                .with_labels(vec![
+                    Label::primary(file_id1, 0..23),
+                    Label::secondary(file_id1, 15..21).with_message("content should be in here"),
+                ]),
+            Diagnostic::error()
+                .with_message("mismatched types")
+                .with_code("E0308")
+                .with_labels(vec![
+                    Label::primary(file_id2, 17..80).with_message("expected (), found integer"),
+                    Label::secondary(file_id2, 55..55).with_message("missing whitespace"),
+                ])
+                .with_notes(vec![
+                    "note:\texpected type `()`\n\tfound type `{integer}`".to_owned()
+                ]),
+        ];
+
+        let test_data = TestData { files, diagnostics };
+
+        insta::assert_snapshot!(test_data.emit_no_color(&config));
```

Comment:
Would it be possible to use the same pattern as the other tests here? ie. doing some setup, and then using a macro to run the test?